lib/jsdoc/util/markdown.js

```diff
@@ -105,6 +105,10 @@ function getParseFunction(parserName, conf) {
         markedRenderer = new marked.Renderer();
 
         markedRenderer.heading = function(text, level) {
```

Comment:
By default, marked's `heading` renderer adds an `id` attribute to headings, so you could simplify things by doing this instead (note that I've changed the option's name as well):

```js
if (!conf.idInHeadings) {
    markedRenderer.heading = function(text, level) {
        // code
    };
}
```